src/ring_theory/perfection.lean

```diff
@@ -151,6 +155,20 @@ any homomorphism `R →+* S` can be lifted to a homomorphism `R →+* perfection
     by rw [← coeff_iterate_frobenius _ 0 n, zero_add, ← ring_hom.map_iterate_frobenius,
       right_inverse_pth_root_frobenius.iterate] }
 
+variables {R p} {S : Type u₂} [comm_semiring S] [char_p S p]
+
+/-- A ring homomorphism `R →+* S` induces `perfection R p →+* perfection S p` -/
+@[simps] def map (φ : R →+* S) : ring.perfection R p →+* ring.perfection S p :=
+{ to_fun := λ f, ⟨λ n, φ (coeff R p n f), λ n, by rw [← φ.map_pow, coeff_pow_p']⟩,
```

Comment:
What is your reason for not defining this in terms of `perfection_map.map`?

Reply:
If we use that then the lemma below wouldn't be rfl I think.

Reply:
Right, that makes sense. But now we have two versions of `map`. So at least we need a lemma that they are equal.

Reply:
Done.